Sources/Utils/POSIX/Daemon.cpp

```diff
@@ -0,0 +1,51 @@
+//
+// Copyright (c) 2014-present, Facebook, Inc.
+// All rights reserved.
+//
+// This source code is licensed under the University of Illinois/NCSA Open
+// Source License found in the LICENSE file in the root directory of this
+// source tree. An additional grant of patent rights can be found in the
+// PATENTS file in the same directory.
+//
+
+#include "DebugServer2/Utils/Daemon.h"
+#include "DebugServer2/Utils/Log.h"
+
+#include <fcntl.h>
+#include <stdio.h>
+#include <unistd.h>
+
+namespace ds2 {
+namespace Utils {
+void Daemonize() {
+  pid_t pid;
+
+  pid = ::fork();
+  if (pid < 0) {
+    DS2LOG(Fatal, "cannot fork: %s", ::strerror(errno));
+  } else if (pid > 0) {
+    ::exit(EXIT_SUCCESS);
+  }
+
+  pid_t sid = ::setsid();
+  if (sid < 0) {
+    DS2LOG(Fatal, "cannot setsid: %s", ::strerror(errno));
+  }
+
+  pid = ::fork();
+  if (pid < 0) {
+    DS2LOG(Fatal, "cannot fork: %s", ::strerror(errno));
+  } else if (pid > 0) {
+    ::exit(EXIT_SUCCESS);
+  }
```

Comment:
@sas explained this to me offline. A good summary is here: https://stackoverflow.com/questions/31485204/why-fork-twice-while-daemonizing